wordpress-mobile/WordPress-iOS
# Approve 2fa challenges via notification action.

This PR modifies how the app handles 2FA auth notifications so an auth challenge may be approved while the app remains in the background. This builds on previous work in #14868 (h/t @jkmassel) and #12216. 

To test:
Apply D52487-code to your wpcom sandbox, and sandbox `wordpress.com` and `public-api.wordpress.com` so you can test with dev/sandboxed push notifications.

Scenario 1:
- Switch to this branch and test the app on a device. 
- Login, view the Notifications tab and allow push notifications.
- Background the app.
- In a browser, attempt to log in to a 2fa enabled account. 
- Confirm you receive a push auth notification.
- Long press on the notification to reveal the approve and deny actions.  Confirm the UI looks good.
- Tap approve.  Confirm the app does not launch.
- Confirm you are logged in. 

Scenario 2:
Launch the app and keep it in the foreground.
- In a browser, attempt to log in to a 2fa enabled account. 
- Confirm that in the app you see a prompt to approve the log in.
- Tap approve.
- Confirm you are logged in. 

@emilylaguna and @jkmassel could I trouble y'all sandbox capable folks for a peek at this one? 
cc @jleandroperez as this might be relevant to your interests

Example:
![push-notification-actions](https://user-images.githubusercontent.com/1435271/98761120-63216200-239a-11eb-9096-b80094c462fc.png)


PR submission checklist:
